attestation-agent/attester/src/nvidia/mod.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2025 Confidential Containers
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::{Attester, TeeEvidence};
+use anyhow::{Context, Result};
+use base64::Engine;
+use nvml_wrapper::{enums::device::DeviceArchitecture, Nvml};
+use serde::Serialize;
+
+pub fn detect_platform() -> bool {
+    // TODO: loop all devices here to check their CC status?
+    match Nvml::init() {
+        Ok(nvml) => nvml
+            .device_count()
+            .map_or_else(|_| false, |count| count > 0),
+        Err(_) => false,
+    }
+}
+
+/// NRAS knows about "switch" and "gpu" but the expected evidence
+/// content is the same. nvidia-attester can compose a list of
+/// all CC enabled nvml devices using this evidence struct.
+#[derive(Serialize)]
+struct NvDeviceReportAndCert {
+    arch: DeviceArchitecture,
+    uuid: String,
+    evidence: String,
+    certificate: String,
+}
+
+#[derive(Serialize)]
+struct NvDeviceEvidence {
+    device_evidence_list: Vec<NvDeviceReportAndCert>,
+}
+
+#[derive(Debug, Default)]
+pub struct NvAttester {}
+
+#[async_trait::async_trait]
+impl Attester for NvAttester {
+    async fn get_evidence(&self, report_data: Vec<u8>) -> Result<TeeEvidence> {
+        let nvml = Nvml::init()?;
+        let devices = nvml.device_count()?;
+
+        let mut device_evidence_list = vec![];
+        let b64_engine = base64::engine::general_purpose::STANDARD;
+
+        for index in 0..devices {
+            let device = nvml.device_by_index(index)?;
+
+            // Take the first half of the reportdata hash as the nonce (the hash includes the KBS protocol nonce)
+            let nonce: [u8; 32] = report_data[0..32].try_into()?;
```

Comment:
Let's add a `warn!` here if the `report_data.len() != 32`. It can be padding or truncated. Also we do not need to mention kbs protocol here as they are decoupled.

Reply:
added but the more fundamental question is can be re-purpose report data for this purpose. my ITA PR shows another problem: it needs to know how nonce became the report data hash to provide it as the "nonce" for the verifier.

Reply:
Only a bytes slice. For devices they do not need to know what the slice mean, only include it exactly same inside the evidence. 

btw, the device attester itself should consider the input slice length and do trimming or padding if needed.